src/main/docker/gitlab/README.md

```diff
@@ -0,0 +1,37 @@
+# GitLab and GitLab CI setup
+
+First, configure the environment parameters:
+```bash
+cp src/main/docker/env.example.gitlab-gitlabci.txt src/main/docker/.env
+vi src/main/docker/.env
+```
+
+Run the following command to start GitLab and a GitLab Runner in a Docker container:
+```bash
+docker-compose -f src/main/docker/gitlab-gitlabci.yml --env-file src/main/docker/.env up --build -d
+```
+
+Then log on to http://localhost/ with the password (`sudo docker exec -it gitlab grep 'Password:' /etc/gitlab/initial_root_password`) and go to http://localhost/admin/runners.
+Click on "Register an instance runner" and copy the registration token.
+Open a shell into the `gitlab-runner` container:
+```bash
+docker exec -it gitlab-runner /bin/bash
+```
+Execute the following command with your token inside the container with the registration token generated above:
+```bash
+gitlab-runner register \
+  --non-interactive \
+  --executor "docker" \
+  --docker-image alpine:latest \
+  --url http://gateway.docker.internal:80 \
+  --clone-url http://gateway.docker.internal:80 \
```

Comment:
```suggestion
  --clone-url http://gitlab:80 \
```

Reply:
As described in the conversation section, the container that is started by the runner and then executes the code is not on the network with the Gitlab instance. Therefore, the URL of the host is needed. For Windows, `gateway.docker.internal` works and for Linux `172.17.0.1`.